examples/gno.land/p/demo/merkle/merkle.gno

```diff
@@ -0,0 +1,134 @@
+package merkle
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"math"
+)
+
+type Hashable interface {
+	Bytes() []byte
+}
+
+type nodes []Node
+
+type Node struct {
+	hash []byte
+
+	position uint8
+}
+
+func (n Node) Hash() string {
+	return hex.EncodeToString(n.hash[:])
+}
+
+type Tree struct {
+	layers []nodes
+}
+
+func (t *Tree) Root() string {
+	for _, l := range t.layers {
+		if len(l) == 1 {
+			return l[0].Hash()
+		}
+	}
+	return ""
+}
+
+func NewTree(data []Hashable) *Tree {
+	tree := &Tree{}
+
+	leaves := make([]Node, len(data))
+
+	tree.layers = make([]nodes, 1)
+	for i, d := range data {
+		hash := sha256.Sum256(d.Bytes())
+
+		tree.layers[0] = append(tree.layers[0], Node{
+			hash: hash[:],
+		})
+
+		leaves[i] = Node{hash: hash[:]}
+	}
```

Comment:
```suggestion
	leaves := make([]Node, len(data))

	for i, d := range data {
		hash := sha256.Sum256(d.Bytes())
		leaves[i] = Node{hash: hash[:]}
	}
	
	tree.layers = []nodes{nodes(leaves)}
```

(this is safe because we don't modify the actual slice in the below code as far as I can see)